stable/external-dns/templates/deployment.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "external-dns.fullname" . }}
+  labels: {{ include "external-dns.labels" . | indent 4 }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels: {{ include "external-dns.labels" . | indent 8 }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8}}
+      {{- end }}
+    spec:
+      {{- if .Values.serviceAccountEnabled }}
+      serviceAccountName: {{ include "external-dns.sa-name" . }}
+      {{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}/{{.Values.image.name}}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args:
+        - --provider={{ .Values.provider }}
+        - --policy={{ .Values.policy }}
+        - --registry={{ .Values.registry }}
+        - --txt-owner-id={{default .Release.Name .Values.txtOwnerId }}
+        {{- if .Values.compatibility }}
```

Comment:
Does each of the following arguments need to have its own value? I'm generally in favor of providing only the arguments that are commonly required to run the application, then providing an `extraArgs` map for users to provide additional ones. See [here](https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/server-deployment.yaml#L51) and [here](https://github.com/kubernetes/charts/blob/master/stable/kube2iam/templates/daemonset.yaml#L36) as examples.

My concern is that the list of options will continue growing, regardless of how commonly-used they are, making the chart unwieldy to new users. Better to keep it simple yet extendable.